### PR TITLE
Document reactor-Core-Micrometer obs/meters/tags via enums

### PR DESCRIPTION
**Pull request by simonbasle**
- add observability doc generation
- Replace constants with PublisherMeters DocumentedMeter
- polish + rewrite the generated meter docs for integration

### CODE REVIEW

**Comment by simonbasle** [comment]
this is an extraction of the resulting docs, as of 1e648c4: 
[Extraction generated metrics docs.pdf](https://github.com/reactor/reactor-core/files/9460525/Extraction.generated.metrics.docs.pdf)

**Comment by simonbasle** [comment]
tentative commit message:
```
This commit uses DocumentedMeter API to replace magic String constants
and make the meters and tags in Reactor-Core-Micrometer documented and
public.

The asciidoc generation now takes advantage of this to generate a
section in the reference guide, through the provided Micrometer tool.

Each of the 3 features (meter listener, observation listener and timed
Scheduler wrapper) get their own DocumenterMeter and one associated
doc generation task. We use a Copy task trick on the three resulting
asciidoc files to remove some of the hardcode boilerplate text for a
better inclusion in the reference guide.

Relates to #3108.
```

**Review by marcingrzejszczak — APPROVED** [review]

File: reactor-core-micrometer/src/main/java/reactor/core/observability/micrometer/MicrometerObservationListener.java [hunk 1]
```diff
@@ -19,13 +19,16 @@
...
 import reactor.core.observability.SignalListener;
+import reactor.core.observability.micrometer.DocumentedObservationListenerTags.ObservationTags;
 import reactor.core.publisher.SignalType;
 import reactor.util.Logger;
 import reactor.util.Loggers;
 import reactor.util.annotation.Nullable;
 import reactor.util.context.Context;
 import reactor.util.context.ContextView;
 
+import static reactor.core.observability.micrometer.DocumentedObservationListenerTags.ObservationTags.*;
```
Comment (chemicL) [inline reactor-core-micrometer/src/main/java/reactor/core/observability/micrometer/MicrometerObservationListener.java:30]: Consider explicit listing of static imports here.

**Review by chemicL — APPROVED** [review]
I noticed in the generated pdf that the javadoc inline html `<p>` is present in the output text, not treated as markup.

**Review by OlegDokuka — APPROVED** [review]

File: reactor-core-micrometer/src/test/java/reactor/core/observability/micrometer/MicrometerObservationListenerConfigurationTest.java [hunk 2]
```diff
@@ -61,7 +61,7 @@ void fromFlux(@Nullable String name, @Nullable String tag) {
 
 		assertThat(configuration.sequenceName)
 			.as("sequenceName")
-			.isEqualTo(name == null ? MicrometerObservationListener.ANONYMOUS_OBSERVATION : name);
+			.isEqualTo(name == null ? "reactor.observation" : name);
```
[the remark this answers is not in the document]
Comment (simonbasle) [reply reactor-core-micrometer/src/test/java/reactor/core/observability/micrometer/MicrometerObservationListenerConfigurationTest.java:64]: yeah that's indeed better

File: reactor-core-micrometer/src/test/java/reactor/core/observability/micrometer/TimedSchedulerTest.java [hunk 3]
```diff
@@ -69,12 +70,12 @@ void constructorAddsDotToPrefixIfNeeded() {
 	}
 
 	@Test
-	void constructorDoesntAddTwoDots() {
-		TimedScheduler test = new TimedScheduler(Schedulers.immediate(), registry, "dot.", Tags.empty());
+	void constructorRejectsDotAtEndOfMetricPrefix() {
```
[the remark this answers is not in the document]
Comment (simonbasle) [reply reactor-core-micrometer/src/test/java/reactor/core/observability/micrometer/TimedSchedulerTest.java:73]: good point. that initially stemmed from trying to use the pattern in the DocumentedMeter, but we can indeed strip the dot beforehand.

File: reactor-core-micrometer/src/main/java/reactor/core/observability/micrometer/MicrometerObservationListener.java (hunk 1 above)
Comment (simonbasle) [reply reactor-core-micrometer/src/main/java/reactor/core/observability/micrometer/MicrometerObservationListener.java:30]: ah I missed that one in my recent commit, 1sec

File: gradle/asciidoc.gradle [hunk 4]
```diff
@@ -74,5 +80,67 @@ configure(rootProject) {
 		}
 		from(asciidoctor) { into("docs/") }
 	}
+
+	configurations {
+		adoc
+	}
+
+	dependencies {
+		//FIXME add to version catalog
```
Comment (simonbasle) [inline gradle/asciidoc.gradle:89]: mmh I should probably address that one right away, and make sure the latest milestone is in use (must be M7 now)

**Review by chemicL — APPROVED** [review]

**Review by chemicL — APPROVED** [review]

**Comment by oscar-picnic** [comment]
Hi @simonbasle I noticed in this PR that for the MicrometerMeterListener metric `%s.flow.duration` you have documented it as a `distribution summary` whereas the actual meter records a `timer`.

Does it make sense to you to change the documentation to a `timer` or am I missing somthing?

**Pull request merged**